src/cnv/ugbio_cnv/process_cnmops_cnvs.py

```diff
@@ -0,0 +1,377 @@
+            )
+            coverage_annotations.append((isamp, oparam, out_annotate_bed_file_cov))
+    return coverage_annotations
+
+
+def add_ids(cnmops_cnv_df: pd.DataFrame) -> pd.DataFrame:
+    """Add IDs to the CNV DataFrame in the format cnmops_<svtype>_<count>.
+
+    Parameters
+    ----------
+    cnmops_cnv_df : pd.DataFrame
+        Input
+
+    Returns
+    -------
+    pd.DataFrame
+        Output, ID added
+    """
+
+    # Add IDs in the format cnmops_<svtype>_<count>
+    svtype_counts = {}
+    ids = []
+    for _, row in cnmops_cnv_df.iterrows():
+        svtype = row["SVTYPE"].lower()
+        svtype_counts[svtype] = svtype_counts.get(svtype, 0) + 1
+        ids.append(f"cnmops_{svtype}_{svtype_counts[svtype]}")
+    cnmops_cnv_df["ID"] = ids
+    return cnmops_cnv_df
+
+
+def run(argv):
+    """
+    Given a bed file, this script will filter it by :
+    1. lcr bed (ug_cnv_lcr) file
+    3. length
+    output consists of 2 files:
+    - VCF file with filtering tags
+    - annotated bed file with filtering tags
+    """
+    parser = argparse.ArgumentParser(
+        prog="process_cnmops_cnvs.py",
+        description="Process CNV calls from cn.mops: filter, annotate, and convert to VCF",
+    )
+    parser.add_argument("--input_bed_file", help="input bed file with .bed suffix", required=True, type=str)
+    parser.add_argument(
+        "--intersection_cutoff",
+        help="intersection cutoff for bedtools subtract function",
+        required=False,
+        type=float,
+        default=0.5,
+    )
+    parser.add_argument("--cnv_lcr_file", help="UG-CNV-LCR bed file", required=False, type=str)
+    parser.add_argument("--min_cnv_length", required=False, type=int, default=10000)
+    parser.add_argument(
+        "--out_directory",
+        help="out directory where intermediate and output files will be saved."
+        " if not supplied all files will be written to current directory",
+        required=False,
+        type=str,
+    )
+    parser.add_argument(
+        "--sample_norm_coverage_file", help="sample normalized coverage file (BED)", required=False, type=str
+    )
+    parser.add_argument("--cohort_avg_coverage_file", help="Cohort average coverage (BED)", required=False, type=str)
+
+    parser.add_argument(
+        "--verbosity",
+        help="Verbosity: ERROR, WARNING, INFO, DEBUG",
+        required=False,
+        default="INFO",
+    )
+    parser.add_argument(
+        "--fasta_index_file",
+        help="tab delimeted file holding reference genome chr ids with their lengths. (.fai file)",
+        required=True,
+        type=str,
+    )
+    parser.add_argument("--sample_name", help="sample name", required=True, type=str)
+
+    args = parser.parse_args(argv[1:])
+    logger.setLevel(getattr(logging, args.verbosity))
+
+    prefix = ""
+    if args.out_directory:
+        prefix = args.out_directory
+        prefix = prefix.rstrip("/") + "/"
+    if "--" in args.input_bed_file:
```

Comment:
maybe we can put this validation within the annotate_bed function since this is a bedtools limitation and related to that code...